.cirrus.yml

```diff
@@ -0,0 +1,227 @@
+  CACHE_PERIOD: "2"
+  # Increment the build number to force new cartopy cache upload.
+  CARTOPY_CACHE_BUILD: "0"
+  # Increment the build number to force new conda cache upload.
+  CONDA_CACHE_BUILD: "0"
+  # Increment the build number to force new nox cache upload.
+  NOX_CACHE_BUILD: "0"
+  # Increment the build number to force new pip cache upload.
+  PIP_CACHE_BUILD: "0"
+  # Pip package to be upgraded/installed.
+  PIP_CACHE_PACKAGES: "pip setuptools wheel nox"
+  # Git commit hash for iris test data.
+  IRIS_TEST_DATA_REF: "fffb9b14b9cb472c5eb2ebb7fd19acb7f6414a30"
+  # Base directory for the iris-test-data.
+  IRIS_TEST_DATA_DIR: ${HOME}/iris-test-data
+
+
+#
+# Linting
+#
+lint_task:
+  auto_cancellation: true
+  name: "Lint: flake8 and black"
+  pip_cache:
+    folder: ~/.cache/pip
+    fingerprint_script:
+      - echo "${CIRRUS_OS} py${PYTHON_VERSION}"
+      - echo "$(date +%Y).$(($(date +%U) / ${CACHE_PERIOD})):${PIP_CACHE_BUILD} ${PIP_CACHE_PACKAGES}"
+  lint_script:
+    - pip list
+    - python -m pip install --retries 3 --upgrade ${PIP_CACHE_PACKAGES}
+    - pip list
+    - nox --session flake8
+    - nox --session black
+
+
+#
+# YAML alias for common linux test infra-structure.
+#
+linux_task_template: &LINUX_TASK_TEMPLATE
+  auto_cancellation: true
+  env:
+    PATH: ${HOME}/miniconda/bin:${PATH}
+    IRIS_REPO_DIR: ${CIRRUS_WORKING_DIR}
+    SITE_CFG: ${CIRRUS_WORKING_DIR}/lib/iris/etc/site.cfg
+  conda_cache:
+    folder: ${HOME}/miniconda
+    fingerprint_script:
+      - wget --quiet https://repo.continuum.io/miniconda/Miniconda3-latest-Linux-x86_64.sh -O miniconda.sh
+      - echo "${CIRRUS_OS} $(sha256sum miniconda.sh)"
+      - echo "$(date +%Y).$(($(date +%U) / ${CACHE_PERIOD})):${CONDA_CACHE_BUILD}"
+    populate_script:
+      - bash miniconda.sh -b -p ${HOME}/miniconda
+      - conda config --set always_yes yes --set changeps1 no
+      - conda config --set show_channel_urls True
+      - conda config --add channels conda-forge
+      - conda update --quiet --name base conda
+      - conda install --quiet --name base nox pip
+  cartopy_cache:
+    folder: ${HOME}/.local/share/cartopy
+    fingerprint_script:
+      - echo "${CIRRUS_OS}"
+      - echo "$(date +%Y).$(($(date +%U) / ${CACHE_PERIOD})):${CARTOPY_CACHE_BUILD}"
+  nox_cache:
+    folder: ${CIRRUS_WORKING_DIR}/.nox
+    fingerprint_script:
+      - echo "${CIRRUS_OS} tests ${PY_VER}"
+      - echo "$(date +%Y).$(($(date +%U) / ${CACHE_PERIOD})):${NOX_CACHE_BUILD}"
+      - sha256sum ${CIRRUS_WORKING_DIR}/requirements/ci/py$(echo ${PY_VER} | tr -d ".").yml
+
+
+#
+# Testing Minimal (Linux)
+#
+linux_minimal_task:
+  matrix:
+    env:
+      PY_VER: 3.6
+    env:
+      PY_VER: 3.7
+  name: "Linux: py${PY_VER} tests (minimal)"
+  container:
+    image: gcc:latest
+    cpu: 2
+    memory: 4G
+  << : *LINUX_TASK_TEMPLATE
+  tests_script:
+    - echo "[Resources]" > ${SITE_CFG}
+    - echo "doc_dir = ${CIRRUS_WORKING_DIR}/docs/iris" >> ${SITE_CFG}
```

Comment:
Note to self - there are a few occurrences of the ``docs`` directory being used for the cirrus-ci setup.